### PR TITLE
Ai make aggressive moves

### PR DESCRIPTION
**Pull request by Cavonstavant**
- Fix unexpected response parsing
- Add different AI modes
    - SURVIVAL
    - AGGRESSIVE
    - DEFENSIVE
- Deny enemy team incantation if possible by going to the incantation location and pushing players

### CODE REVIEW

**Review by 31Nathan — APPROVED** [review]
Very cool, especially the AI Mode Class

File: src/zappy_ai_src/ai_strategy_management.py [hunk 1]
```diff
@@ -437,21 +441,20 @@ def __unexpectedResponseManagement(self):
         """
         print("RENTRE DANS UNEXPECTED RESPONSE MANAGEMENT")
         response: str = self.__lib.getUnexpectedResponse()
-        if response == "":
+        if response == "" or response != None:
```
Comment (LucasTesnier) [inline src/zappy_ai_src/ai_strategy_management.py:444]: != None ? Always true

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Just answer my comment

File: src/zappy_ai_src/ai_strategy_management.py (hunk 1 above)
Comment (Cavonstavant) [reply src/zappy_ai_src/ai_strategy_management.py:444]: this is just a preventive check, nothing crazy don't worry

**Pull request merged**